src/ameba/rule/lint/whitespace_around_macro_expression.cr

```diff
@@ -0,0 +1,50 @@
+module Ameba::Rule::Lint
+  # A rule that checks for whitespace around macro expressions.
+  #
+  # This is considered invalid:
+  #
+  # ```
+  # {{foo}}
+  # ```
+  #
+  # And it has to written as this instead:
+  #
+  # ```
+  # {{ foo }}
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Lint/WhitespaceAroundMacroExpression:
+  #   Enabled: true
+  # ```
+  class WhitespaceAroundMacroExpression < Base
+    include AST::Util
+
+    properties do
+      since_version "1.7.0"
+      description "Reports missing spaces around macro expressions"
+    end
+
+    MSG = "Missing spaces around macro expression"
+
+    def test(source, node : Crystal::MacroExpression)
+      # https://github.com/crystal-lang/crystal/pull/15524
+      {% if compare_versions(Crystal::VERSION, "1.16.0") < 0 %}
+        return
+      {% end %}
+
+      return unless node.output?
+      return unless code = node_source(node, source.lines)
+      return if code.starts_with?("{{ ") && code.ends_with?(" }}")
+
+      issue_for node, MSG do |corrector|
+        corrected_code =
+          "{{ #{code.lstrip('{').rstrip('}').strip(' ')} }}"
```

Comment:
(minor) from a performance perspective, this is creating a lot of intermediate strings that are immediately thrown away. May be better to use a string builder and iterate over the chars of the code string. May be worth combing thru the codebase for any other low hanging fruit like this.

Reply:
Should also only remove the first and last two brackets, so it could be something like this:

```
corrected_code = String.build do |bld|
  bld << "{{ "

  code.each_char_with_index do |char, idx|
    # Skip brackets
    next if idx < 2 || idx > (code.size - 3)

    # rest of handling char
  end

  bld << " }}"
end
```